docs/DeveloperGuide.md

```diff
@@ -1,34 +1,247 @@
+ **Cons:** Activity names have to be unique.
+ **Pros:** More versatile, resistant to changes in the activity list
+ - Continue by activity index
+ **Cons:** need to add an additional index field to the Activity class, 
+ index is not fixed, changes when an activity is deleted
+ **Pros:** Can reuse activity names.
+ 
+Although the current implementation of the continue feature disallows users to have multiple activities with the same name, we felt that the versatility of this choice outweighed the cons. Firstly because if the activityList got too big, it would be hard for the user to get the index of the task he/she wanted to continue. Also, the index would constantly be changing when changes are made to the list.
+
+#### 3.5.3 Additional Features
+As users can only have activities with unique names, when a user wants to start an activity which already exists in the activityList, he/she will be given the option to continue the stated activity.
+![decision flowchart](./pictures/continue_flowchart.PNG)
+
+### 3.6 Find Feature
+This command accepts a keyword and searches the activity list for activities with names that contain the keyword.
+
+#### 3.6.1 Current Implementation
+* This feature is called by the user when the “find” command is entered into the command line. The string following the command is the keyword to match activity names to.
+* The Parser will create a FindCommand object.
+* The FindCommand will invoke its own executeCommand() method.
+    * The Parser's lastShownList will be cleared.
+    * Then it will loop through activityList to find activities with names that contain the keyword.
+    * If one is found, it will be added to lastShownList.
+    * printResults() of the Ui will be called:
+        * If lastShownList is not empty, it will print the matching activities.
+        * Else, it will respond to the user that there are no tasks which match the given keyword.
+
+
+![find seq diagram](https://imgur.com/Icg5rdB.png)
+### 3.7 Filter Feature
+This feature accepts multiple space-separated keywords to search for activities with tags matching each keyword.
+
+#### 3.7.1 Current Implementation
+* This feature is called by the user when the “filter” command is entered into the command line. The space separated strings following the command are the keywords to match activity tags with.
+* The Parser will create a FilterCommand object.
+* The FindCommand will invoke its own executeCommand() method.
+* The Parser's lastShownList will be cleared.
+* For each keyword:
+    * Then it will loop through activityList to find activities with tags that contain the keyword.
+    * If one is found, it will be added to lastShownList.
+    * printResults() method of the Ui will be called
+        * If lastShownList is not empty, it will print the matching activities.
+        * Else, it will respond to the user that there are no tasks which match the given keyword.
+
+
+![filter seq diagram](https://imgur.com/hybT3R9.png)
```

Comment:
The diagram that is drawn may be too complicated for the readers as there are many "Loop" and "opt" between the Parser class and the FilterCommand class. Perhaps, you could consider breaking down the diagrams to show smaller components so that it is easier to understand?